### PR TITLE
Move tests to use the emulator

### PR DESCRIPTION
**Pull request by samtstern**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -11,9 +11,19 @@ else
...
+
 # Only run test suite when we can decode the service acct
 if [ "$TRAVIS_SECURE_ENV_VARS" = false ]; then
   echo "Could not find secure environment variables, skipping integration tests."
 else
-  GOOGLE_APPLICATION_CREDENTIALS=service-account.json npm run build && npm run test
+  $FIREBASE --open-sesame emulators
+  $FIREBASE setup:emulators:firestore
+  $FIREBASE serve --only firestore &
+  sleep 5
```
Comment (scottcrossen) [inline build.sh]: Consider using nc as done here:
https://github.com/firebase/InternalProjects/blob/master/2018-10-16-piglatin-firestore/rules-testing/run-tests.sh

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,5901 @@
+{
```
Comment (scottcrossen) [inline package-lock.json]: This seems like a good thing to add to .gitignore

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,6 @@
 service-account.json
 node_modules
 dist/
+
+firebase-debug.log
+firestore-debug.log
```
Comment (scottcrossen) [inline .gitignore]: Github says you don't have a newline here

File: build.sh (hunk 1 above)
Comment (samtstern) [reply build.sh]: Good call

File: package-lock.json (hunk 2 above)
Comment (samtstern) [reply package-lock.json]: It's noisy but they recommend committing it

**Pull request merged**